### ISSUE TITLE
on restarting rqlited service, it shows earlier dropped table!

### ISSUE THREAD

**Issue by limenleap** [rqlite/rqlite#775]
I installed rqlite v5.10.2 at an Amazon lightsail 0.5GB Ubuntu server.
I gave a JSON configuration file so that only one user has "ALL" access on the http API 
Did all the firewall setting at lightsail for 4001 and 4002
Wrote a simple systemctld service script for easy start, stop and reset

I created a table (lets call  it  "tablea") with just a couple of columns. Then added a sample record. All that was being done from my Windows computer at home via the HTTP api

Everything fine till now!

Then I realized that i needed a different kind of table. So I dropped the table again using the HTTP api  from my windows computer. 
Added a new table with the various columns I really needed. I called this one also "tablea" as before 
Also, made a configuration change: This time I used the -on-disk option also; so a SQLITE file was created 
Added some sample records into the newly created table 
I queried the table again using HTTP API -- all well!

In short, everything went smooth
Till I restarted the service at the Ubuntu Lightsail server 

Magically, it is now showing the FIRST table version  -- there is absolutely no sign of the revised table. I am still getting the very first version; the one that I had created without the _-on-disk_ option with the original sample records!!

**Comment by otoolep** [comment]
Thanks for your report. Can you please provide more details?: I need very clear details on what you are doing, and the sequencing. This means showing me exactly the commands you use to launch rqlited each time, how you shut it down, and the HTTP commands you execute.

I need the following, in a clear sequence

1. The command you run to launch your brand new node.
2. The HTTP commands you actually execute, show me via `curl -v`.
3. what you do next
4. the command you execute to restart your node
5. how you confirm the table is still there. Again, show me `curl -v` hitting the API, and the request that comes back.

**Comment by limenleap** [comment]
OOps... I deleted the data folder and remade it. Sorry but I can give some more details 

Here is the bash script that I had used (In the first version; the -on-disk option was not present. No other change)

```
#! /usr/bin/bash

/apps/rqlite/rqlitebin/rqlited -auth /apps/rqlite/limenkoyns/limenkoyns.config.json -on-disk -http-cert /apps/rqlite/limenkoyns/cert.pem -http-key /apps/rqlite/limenkoyns/key.pem  -node-cert  /apps/rqlite/limenkoyns/cert.pem  -node-key /apps/rqlite/limenkoyns/key.pem -node-id 1 -http-addr 0.0.0.0:4001 -raft-addr 0.0.0.0:4002  -node-no-verify /apps/rqlite/limenkoyns/data
```

Here is the systemctld script to sstart, stop and reset the service

```

[Unit]
Description=RQlite Service For Limen Leap Labs Koyns
After=network.target
StartLimitIntervalSec=0
[Service]
Type=simple
Restart=always
RestartSec=1
User=ubuntu
ExecStart=/usr/bin/env bash /apps/rqlite/limenkoyns/startLimenKoyns.sh

[Install]
WantedBy=multi-user.target

```
Here is the "magical" output from the FIRST table, even though it was dropped (The username and password is hidden)

![image](https://user-images.githubusercontent.com/20351066/110013760-0b8f7380-7d48-11eb-8d59-5f1472196586.png)

After that I dropped the table once again; recreated the same table once again with new columns as seen below

![image](https://user-images.githubusercontent.com/20351066/110014098-6a54ed00-7d48-11eb-9e43-f1b6af4136cf.png)

Now in this following image, you can see the actual problem.  I had inserted some test record into the newly defined table structure -- and you can see a bit of the test at the top of the image. You wold see that in the JSON, the "values": array has 1 record with the values of 4 columns .

After that i reset the service at the server; and then executed the curl from Windows -- and magically, it is now displaying the very first in-memory table (You can see the  "values" array has 2 records, and each record has only 3 columns. 


![image](https://user-images.githubusercontent.com/20351066/110014535-e94a2580-7d48-11eb-9ecb-b0f8da8b9029.png)

**Comment by limenleap** [comment]
In my desperation to solve this problem; I actually stopped the rqlited service, deleted the data folder; recreated it again, rebooted the server. And now it seems to behave as expected. 

Pardon me -- I think I must have goofed up in not preserving the state. I hope whatever info I did provide would help you. I am grateful for rqlite  I think it solves a big problem and I will try and help as much as I can

**Comment by otoolep** [comment]
I'll try to reproduce, but if I cannot I will close this issue.

By the way, screenshots, while they seem informative, are awkward to handle. I cannot copy and paste. If it better to copy from your terminal and paste the text directly into the ticket.

**Comment by limenleap** [comment]
Here are the copy pasted data from Windows (I've removed the username and password) 

This was the first surprise today because yesterday I had changed the table. But today; after I restarted with the -on-disk setting, I got the earlier Table back

```
curl -k -G "https://65.1.137.196:4001/db/query?pretty&timings" --data-urlencode "q=SELECT * FROM koyns"
{
    "results": [
        {
            "columns": [
                "id",
                "koyncode",
                "koynvalue"
            ],
            "types": [
                "integer",
                "text",
                "char[255]"
            ],
            "values": [
                [
                    1,
                    "genesis",
                    "TheFirst"
                ],
                [
                    2,
                    "K1614796111",
                    "TheSecond"
                ]
            ],
            "time": 0.000125986
        }
    ],
    "time": 0.000147672
}
```

I then dropped the table:

```
curl -k -G "https://65.1.137.196:4001/db/query" --data-urlencode "q=DROP TABLE koyns"
{"results":[{}]}

```

Then I verified if the table existed. It indicated no such table 

```
curl -k -G "https://65.1.137.196:4001/db/query?pretty&timings" --data-urlencode "q=SELECT * FROM koyns"
{
    "results": [
        {
            "error": "no such table: koyns"
        }
    ],
    "time": 0.000124321
}

```

Then I created the new table 

```
curl -k -X POST "https://65.1.137.196:4001/db/execute?pretty&timings" -H "Content-Type:application/json" -d "[\"CREATE TABLE koyns (id INTEGER NOT NULL PRIMARY KEY, koyntime TEXT, koyncode CHAR[255], koyninfo TEXT, reused INTEGER NOT NULL)\"]"
{
    "results": [
        {
            "time": 0.004323304
        }
    ],
    "time": 0.006093542
}

```
Then I verified if the new table indeed had the new structure. Yes, it had

```
curl -k -G "https://65.1.137.196:4001/db/query?pretty&timings" --data-urlencode "q=SELECT * FROM koyns"
{
    "results": [
        {
            "columns": [
                "id",
                "koyntime",
                "koyncode",
                "koyninfo",
                "reused"
            ],
            "types": [
                "integer",
                "text",
                "char[255]",
                "text",
                "integer"
            ],
            "time": 0.000121268
        }
    ],
    "time": 0.000144805
}

```
Then I reset the service and checked again. This time it magically showed the first table, with the initial data I had inserted in there

```
>curl -k -G "https://65.1.137.196:4001/db/query?pretty&timings" --data-urlencode "q=SELECT * FROM koyns"
{
    "results": [
        {
            "columns": [
                "id",
                "koyncode",
                "koynvalue"
            ],
            "types": [
                "integer",
                "text",
                "char[255]"
            ],
            "values": [
                [
                    1,
                    "genesis",
                    "TheFirst"
                ],
                [
                    2,
                    "K1614796111",
                    "TheSecond"
                ]
            ],
            "time": 0.000176778
        }
    ],
    "time": 0.000204934
}

```

Did some more tests  -- including adding some data to the new table, etc. But after reset; it went back to the first table. Finally; as explained earlier, I got desperate -- so I deleted the data directory on the Linux server, rebooted it, etc. Now it seems to be stable

**Comment by otoolep** [comment]
I cannot reproduce this. There is also a fair amount of automated testing in this area.

If you can come up with a repro case, please re-open.

**Comment by orzFly** [comment]
```curl -k -G "https://65.1.137.196:4001/db/query" --data-urlencode "q=DROP TABLE koyns"```

It seems that you should never drop a table with **query**. Use **execute** instead.

**Comment by otoolep** [comment]
Yes, important to use the right endpoint. 6.1.0 will stop this happening anyway, as the query endpoint uses a read-only connection to the database.
